### ISSUE TITLE
[java] UnusedPrivateField: False positive on JUnit 5/6 @FieldSource

### ISSUE THREAD

**Issue by UncleOwen** [pmd/pmd#6606]
**Affects PMD Version:** 7.23.0

**Rule:** [UnusedPrivateField](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedprivatefield)

**Description:** When a private field is used as an argument source for a [@ParameterizedTest](https://docs.junit.org/6.0.3/writing-tests/parameterized-classes-and-tests.html#sources-FieldSource), it is reported as unused by PMD.

**Code Sample demonstrating the issue:**

```java
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.FieldSource;

import java.util.List;

class FooTest {
    private static final List<String> foo = List.of("a");

    @ParameterizedTest
    @FieldSource("foo")
    void testFoo(String arg) {}
}

```

**Expected outcome:**

PMD reports a violation at line 7, but that's wrong. That's a false positive.

**Running PMD through:** CLI
